File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.go
func maxScore(s string) int {
    right := 0
    maxScore := 0

    for i := 1; i < len(s); i++ {
        if string(s[i]) == "1" {
            right++
        }
    }

    maxScore = right
    left := 0

    for i := 0; i < len(s) - 1; i++ {
        if string(s[i]) == "0" {
            left++
        } else if string(s[i]) == "1" {
            if i == 0 {
                continue
            }
            right--
        }
        if left + right > maxScore {
            maxScore = left + right
        }
    }
    return maxScore
}

File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.cpp
auto init = [](){
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    return 'c';
}();
class Solution {
public:
    const int maxScore(const string s) {
        int result = 0;
        int maxRight = 0;
        int maxLeft = 0;
        for (const char& c : s) if (c == '1') ++maxRight;
        for (int i = 0; i < s.size() - 1; ++i) {
            if (s[i] == '1') --maxRight;
            else ++maxLeft;
            result = max(result, maxLeft + maxRight);
        }
        return result;
    }
};